model/metrics/gen-ai.yaml

```diff
@@ -43,3 +55,27 @@ groups:
           The `error.type` SHOULD match the error code returned by the Generative AI provider or the client library,
           the canonical name of exception that occurred, or another low-cardinality error identifier.
           Instrumentations SHOULD document the list of errors they report.
+  - id: metric.gen_ai.server.request.duration
```

Comment:
Was it considered that we will have two duration histograms, when these http calls are [also instrumented ](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#metric-httpserverrequestduration)? I assume since here there's extra attributes this is fine, but just wanted to clarify.

Reply:
It's a good point.  However - I think we have this general problem around  "layers" of instrumentation.   Ideally - users would have a good default, but they can pick which layers to keep and which to drop.  My ideal default would be redunandant instrumentation would always drop the "furthest semantic meaning" -i.e. if I have HTTP request metrics *AND* AI/ML - the AI/ML metrics are closer in semantic to what I'm doing so by default they'd be preserved.

I believe @lmolkova has a write-up on this issue somewhere in the specification at some point.   For now, I think this is ok.   The need for users to be able to control semantic-layers of information and the fact that we'll have overlap conflicts like this is not new, nor is this PR the first to cause the issue.

Reply:
On the client side we separate all 'logical' from all 'physical' because of the retries, redirect, auth, response streaming. 
It's slightly different on the server.

@achandrasekar wonder what your thoughts are - what are the instrumentation point for LLM metrics? are they coming from the model or from the HTTP server?  
I assume it could be possible to use LLM with a different protocol, would the same LLM metrics apply/make sense?

Also, echoing @jsuereth comment I'd assume LLM servers would not expose HTTP metrics if they are quite similar to HTTP. If they are different, then reporting both could make sense.

Reply:
This server metric is coming from the model server. And there can be different protocols supported by the model server - gRPC is a popular one that model servers support in addition to HTTP. In some cases, gRPC is the only one supported by default and a separate HTTP FastAPI server is provided as an additional component that users can deploy if needed. So, I think having a model server level "request duration" different from the HTTP one makes sense to report this along with the other model server level latency metrics like time to first token.

Reply:
Thanks for the insights, makes sense to me!